cmd/agent/app/reporter/grpc/reporter.go

```diff
@@ -34,20 +34,22 @@ type Reporter struct {
 	collector api_v2.CollectorServiceClient
 	logger    *zap.Logger
 	sanitizer zipkin2.Sanitizer
+	agentTags []model.KeyValue
 }
 
 // NewReporter creates gRPC reporter.
-func NewReporter(conn *grpc.ClientConn, logger *zap.Logger) *Reporter {
+func NewReporter(conn *grpc.ClientConn, agentTags string, logger *zap.Logger) *Reporter {
 	return &Reporter{
 		collector: api_v2.NewCollectorServiceClient(conn),
 		logger:    logger,
 		sanitizer: zipkin2.NewChainedSanitizer(zipkin2.StandardSanitizers...),
+		agentTags: parseAgentTags(agentTags),
```

Comment:
mixing concern: parsing of tags string should be done wherever CLI flags are parsed. Reporter should take `map[string]string` or even better `[]model.Tag`